src/spec-configuration/containerFeaturesConfiguration.ts

```diff
@@ -5,20 +5,32 @@
 
 import * as jsonc from 'jsonc-parser';
 import * as path from 'path';
-import * as semver from 'semver';
+//import * as semver from 'semver';
 import * as URL from 'url';
 import * as tar from 'tar';
 import { DevContainerConfig } from './configuration';
-import { mkdirpLocal, readLocalFile, rmLocal, writeLocalFile } from '../spec-utils/pfs';
+import { mkdirpLocal, readLocalFile, rmLocal, writeLocalFile, cpDirectoryLocal } from '../spec-utils/pfs';
 import { Log, LogLevel } from '../spec-utils/log';
 import { request } from '../spec-utils/httpRequest';
+import { DevContainerFeature } from './configuration';
+import { existsSync } from 'fs';
 
 const ASSET_NAME = 'devcontainer-features.tgz';
 
 export interface Feature {
 	id: string;
 	name: string;
+	description?: string;
+	filename?: string;
+	runApp?: string;
+	runParams?: string;
+	infoString?: string;
+	internalVersion?: string;
+	tempLocalPath?: string;
```

Comment:
It would be nice to know here which properties can be user-supplied in `devcontainer-feature.json`, and which are `// set programatically`